### PR TITLE
OP-904 Change date chooser on Pharmacy->Pharmaceuticals Stock Ward

### PR DESCRIPTION
**Pull request by dbmalkovsky**
See OP-904

### CODE REVIEW

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java [hunk 1]
```diff
@@ -447,34 +445,34 @@ private Component getJPanelRange() {
...
-			jCalendarTo.addPropertyChangeListener("date", propertyChangeEvent -> {
-				dateTo = Converters.convertToLocalDateTime((Date) propertyChangeEvent.getNewValue());
-				jTableOutcomes.setModel(new OutcomesModel());
-				jTableIncomes.setModel(new IncomesModel());
-				rowCounter.setText(rowCounterText + jTableOutcomes.getRowCount());
+			jCalendarTo = new GoodDateChooser(dateTo.toLocalDate(), false);
+			jCalendarTo.addDateChangeListener(dateChangeEvent -> {
+				LocalDate newDate = dateChangeEvent.getNewDate();
+				if (newDate != null) {
+					dateTo = newDate.atStartOfDay(); // TODO
```
Comment (mwithi) [inline src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java:454]: end of the day here?

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java [hunk 2]
```diff
@@ -447,34 +445,34 @@ private Component getJPanelRange() {
...
-			jCalendarFrom.addPropertyChangeListener("date", propertyChangeEvent -> {
-				dateFrom = Converters.convertToLocalDateTime((Date) propertyChangeEvent.getNewValue());
-				jTableOutcomes.setModel(new OutcomesModel());
-				jTableIncomes.setModel(new IncomesModel());
-				rowCounter.setText(rowCounterText + jTableOutcomes.getRowCount());
+			jCalendarFrom = new GoodDateChooser(dateFrom.toLocalDate());
+			jCalendarFrom.addDateChangeListener(dateChangeEvent -> {
+				LocalDate newDate = dateChangeEvent.getNewDate();
+				if (newDate != null) {
+					dateFrom = newDate.atStartOfDay(); // TODO
```
Comment (mwithi) [inline src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java:471]: TODO comment is for what?

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java [hunk 3]
```diff
@@ -1675,8 +1673,8 @@ private String compileFileName() {
 
 			filename.append("_").append(jComboBoxMedicals.getSelectedItem());
 		}
-		filename.append("_").append(TimeTools.formatDateTime(jCalendarFrom.getLocalDateTime(), DATE_FORMAT_YYYYMMDD))
-				.append("_").append(TimeTools.formatDateTime(jCalendarTo.getLocalDateTime(), DATE_FORMAT_YYYYMMDD));
+		filename.append("_").append(TimeTools.formatDateTime(jCalendarFrom.getDateStartOfDay(), DATE_FORMAT_YYYYMMDD))
+				.append("_").append(TimeTools.formatDateTime(jCalendarTo.getDateStartOfDay(), DATE_FORMAT_YYYYMMDD));
```
Comment (mwithi) [inline src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java:1677]: end of the day here, but it is irrilevant for the date format I guess

**Review by mwithi — CHANGES_REQUESTED** [review]
LGTM, but little changes

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java (hunk 1 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java:454]: yes; fixed

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java (hunk 2 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java:471]: removed; was a mistake

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java (hunk 3 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java:1677]: fixed it anyway

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstockward/gui/WardPharmacy.java:454]: the comment too I guess

**Comment by dbmalkovsky** [comment]
All comments addressed I think.

**Review by mwithi — APPROVED** [review]

**Pull request merged**